In [3]:

#library



import pandas as pd 
import numpy as np

pd.options.display.max_rows=100




#- ETUDE DES VALEURS MANQUANTES :
#- REPARTITION DES MODALITés :
#- Stat bivarie 
#- Matrice de corrélation (variables retenues)

In [4]:
#importation des données 
test_df = pd.read_csv('//ad.univ-lille.fr/Etudiants/Homedir3/141794/Documents/SIAD/M2 SIAD/S9/projet BigData/test.csv')
train_df = pd.read_csv('//ad.univ-lille.fr/Etudiants/Homedir3/141794/Documents/SIAD/M2 SIAD/S9/projet BigData/train.csv')




In [5]:
# Vue globale sur les données
train_df.columns
test_df.columns
train_df.info()
test_df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 81 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id             1460 non-null   int64  
 1   MSSubClass     1460 non-null   int64  
 2   MSZoning       1460 non-null   object 
 3   LotFrontage    1201 non-null   float64
 4   LotArea        1460 non-null   int64  
 5   Street         1460 non-null   object 
 6   Alley          91 non-null     object 
 7   LotShape       1460 non-null   object 
 8   LandContour    1460 non-null   object 
 9   Utilities      1460 non-null   object 
 10  LotConfig      1460 non-null   object 
 11  LandSlope      1460 non-null   object 
 12  Neighborhood   1460 non-null   object 
 13  Condition1     1460 non-null   object 
 14  Condition2     1460 non-null   object 
 15  BldgType       1460 non-null   object 
 16  HouseStyle     1460 non-null   object 
 17  OverallQual    1460 non-null   int64  
 18  OverallC

Comme nous pouvons le constater, les variables Alley, FireplaceQu, PoolQC, Fence, MiscFeature  ne contiennent pas assez de modalités. Nous avons donc decidé de les supprimer 

In [6]:
# vérifier s'il ya de lignes qui sont dupliquées 
train_df.duplicated().sum()
test_df.duplicated().sum()

0

Nous n'avons aucune ligne dupliquée

Traitement des valeurs manquantes

In [7]:
#valeurs manquantes dans df-train
percent_missing_train = train_df.isnull().sum() * 100 / len(train_df)
missing_value_df_train = pd.DataFrame({'Colonne': train_df.columns,
                                 'Pourcentage': percent_missing_train})


missing_value_df_train.sort_values('Pourcentage', inplace=True)
missing_value_df_train

Colonne  Pourcentage
Id                        Id     0.000000
TotalBsmtSF      TotalBsmtSF     0.000000
Heating              Heating     0.000000
SaleCondition  SaleCondition     0.000000
CentralAir        CentralAir     0.000000
1stFlrSF            1stFlrSF     0.000000
2ndFlrSF            2ndFlrSF     0.000000
LowQualFinSF    LowQualFinSF     0.000000
GrLivArea          GrLivArea     0.000000
BsmtFullBath    BsmtFullBath     0.000000
BsmtHalfBath    BsmtHalfBath     0.000000
FullBath            FullBath     0.000000
HalfBath            HalfBath     0.000000
BedroomAbvGr    BedroomAbvGr     0.000000
KitchenAbvGr    KitchenAbvGr     0.000000
KitchenQual      KitchenQual     0.000000
TotRmsAbvGrd    TotRmsAbvGrd     0.000000
Functional        Functional     0.000000
SaleType            SaleType     0.000000
YrSold                YrSold     0.000000
MoSold                MoSold     0.000000
MiscVal              MiscVal     0.000000
PoolArea            PoolArea     0.000000
ScreenPorch      ScreenPorch     0.000000
BsmtUnfSF          BsmtUnfSF     0.000000
3SsnPorch          3SsnPorch     0.000000
OpenPorchSF      OpenPorchSF     0.000000
WoodDeckSF        WoodDeckSF     0.000000
PavedDrive        PavedDrive     0.000000
GarageArea        GarageArea     0.000000
GarageCars        GarageCars     0.000000
Fireplaces        Fireplaces     0.000000
EnclosedPorch  EnclosedPorch     0.000000
BsmtFinSF2        BsmtFinSF2     0.000000
HeatingQC          HeatingQC     0.000000
SalePrice          SalePrice     0.000000
LandSlope          LandSlope     0.000000
LandContour      LandContour     0.000000
LotShape            LotShape     0.000000
Neighborhood    Neighborhood     0.000000
Condition1        Condition1     0.000000
Street                Street     0.000000
LotArea              LotArea     0.000000
MSZoning            MSZoning     0.000000
BsmtFinSF1        BsmtFinSF1     0.000000
Condition2        Condition2     0.000000
BldgType            BldgType     0.000000
HouseStyle        HouseStyle     0.000000
OverallQual      OverallQual     0.000000
OverallCond      OverallCond     0.000000
LotConfig          LotConfig     0.000000
YearBuilt          YearBuilt     0.000000
RoofStyle          RoofStyle     0.000000
RoofMatl            RoofMatl     0.000000
Exterior1st      Exterior1st     0.000000
Exterior2nd      Exterior2nd     0.000000
ExterQual          ExterQual     0.000000
ExterCond          ExterCond     0.000000
Foundation        Foundation     0.000000
MSSubClass        MSSubClass     0.000000
YearRemodAdd    YearRemodAdd     0.000000
Utilities          Utilities     0.000000
Electrical        Electrical     0.068493
MasVnrType        MasVnrType     0.547945
MasVnrArea        MasVnrArea     0.547945
BsmtQual            BsmtQual     2.534247
BsmtCond            BsmtCond     2.534247
BsmtFinType1    BsmtFinType1     2.534247
BsmtFinType2    BsmtFinType2     2.602740
BsmtExposure    BsmtExposure     2.602740
GarageQual        GarageQual     5.547945
GarageFinish    GarageFinish     5.547945
GarageYrBlt      GarageYrBlt     5.547945
GarageType        GarageType     5.547945
GarageCond        GarageCond     5.547945
LotFrontage      LotFrontage    17.739726
FireplaceQu      FireplaceQu    47.260274
Fence                  Fence    80.753425
Alley                  Alley    93.767123
MiscFeature      MiscFeature    96.301370
PoolQC                PoolQC    99.520548

In [8]:
#valeurs manquantes dans df-test
percent_missing_test = test_df.isnull().sum() * 100 / len(test_df)
missing_value_df_test = pd.DataFrame({'Colonne': test_df.columns,
                                 'Pourcentage': percent_missing_test})


missing_value_df_test.sort_values('Pourcentage', inplace=True)
missing_value_df_test

Colonne  Pourcentage
Id                        Id     0.000000
Foundation        Foundation     0.000000
HeatingQC          HeatingQC     0.000000
CentralAir        CentralAir     0.000000
Electrical        Electrical     0.000000
1stFlrSF            1stFlrSF     0.000000
2ndFlrSF            2ndFlrSF     0.000000
LowQualFinSF    LowQualFinSF     0.000000
GrLivArea          GrLivArea     0.000000
FullBath            FullBath     0.000000
HalfBath            HalfBath     0.000000
BedroomAbvGr    BedroomAbvGr     0.000000
ExterCond          ExterCond     0.000000
KitchenAbvGr    KitchenAbvGr     0.000000
Fireplaces        Fireplaces     0.000000
PavedDrive        PavedDrive     0.000000
WoodDeckSF        WoodDeckSF     0.000000
OpenPorchSF      OpenPorchSF     0.000000
EnclosedPorch  EnclosedPorch     0.000000
3SsnPorch          3SsnPorch     0.000000
ScreenPorch      ScreenPorch     0.000000
PoolArea            PoolArea     0.000000
MiscVal              MiscVal     0.000000
MoSold                MoSold     0.000000
YrSold                YrSold     0.000000
TotRmsAbvGrd    TotRmsAbvGrd     0.000000
ExterQual          ExterQual     0.000000
Heating              Heating     0.000000
SaleCondition  SaleCondition     0.000000
OverallQual      OverallQual     0.000000
OverallCond      OverallCond     0.000000
YearBuilt          YearBuilt     0.000000
YearRemodAdd    YearRemodAdd     0.000000
RoofStyle          RoofStyle     0.000000
RoofMatl            RoofMatl     0.000000
Condition2        Condition2     0.000000
Condition1        Condition1     0.000000
BldgType            BldgType     0.000000
Neighborhood    Neighborhood     0.000000
LotConfig          LotConfig     0.000000
LandContour      LandContour     0.000000
LotShape            LotShape     0.000000
Street                Street     0.000000
LotArea              LotArea     0.000000
MSSubClass        MSSubClass     0.000000
LandSlope          LandSlope     0.000000
HouseStyle        HouseStyle     0.000000
GarageCars        GarageCars     0.068540
GarageArea        GarageArea     0.068540
KitchenQual      KitchenQual     0.068540
Exterior1st      Exterior1st     0.068540
SaleType            SaleType     0.068540
TotalBsmtSF      TotalBsmtSF     0.068540
BsmtUnfSF          BsmtUnfSF     0.068540
Exterior2nd      Exterior2nd     0.068540
BsmtFinSF1        BsmtFinSF1     0.068540
BsmtFinSF2        BsmtFinSF2     0.068540
BsmtFullBath    BsmtFullBath     0.137080
Functional        Functional     0.137080
Utilities          Utilities     0.137080
BsmtHalfBath    BsmtHalfBath     0.137080
MSZoning            MSZoning     0.274160
MasVnrArea        MasVnrArea     1.028101
MasVnrType        MasVnrType     1.096642
BsmtFinType2    BsmtFinType2     2.878684
BsmtFinType1    BsmtFinType1     2.878684
BsmtExposure    BsmtExposure     3.015764
BsmtQual            BsmtQual     3.015764
BsmtCond            BsmtCond     3.084304
GarageType        GarageType     5.209047
GarageFinish    GarageFinish     5.346127
GarageYrBlt      GarageYrBlt     5.346127
GarageQual        GarageQual     5.346127
GarageCond        GarageCond     5.346127
LotFrontage      LotFrontage    15.558602
FireplaceQu      FireplaceQu    50.034270
Fence                  Fence    80.123372
Alley                  Alley    92.666210
MiscFeature      MiscFeature    96.504455
PoolQC                PoolQC    99.794380

Liste des variables manquantes ayant 
TEST
GarageArea	GarageArea	0.068540
KitchenQual	KitchenQual	0.068540
GarageCars	GarageCars	0.068540
TotalBsmtSF	TotalBsmtSF	0.068540
Exterior1st	Exterior1st	0.068540
Exterior2nd	Exterior2nd	0.068540
BsmtFinSF1	BsmtFinSF1	0.068540
BsmtFinSF2	BsmtFinSF2	0.068540
SaleType	SaleType	0.068540
BsmtUnfSF	BsmtUnfSF	0.068540
BsmtFullBath	BsmtFullBath	0.137080
Functional	Functional	0.137080
Utilities	Utilities	0.137080
BsmtHalfBath	BsmtHalfBath	0.137080
MSZoning	MSZoning	0.274160
MasVnrArea	MasVnrArea	1.028101
MasVnrType	MasVnrType	1.096642

TRAIN
Electrical	Electrical	0.068493
MasVnrArea	MasVnrArea	0.547945
MasVnrType	MasVnrType	0.547945


Comme nous pouvons le constater, les variables Alley, FireplaceQu, PoolQC, Fence, MiscFeature ne contiennent pas assez de modalités. En effet, ces variables ont plus de 50% de données manquantes. Nous avons donc decidé de les supprimé

In [9]:
train_df = train_df.drop(['Alley','PoolQC', 'Fence', 'MiscFeature','FireplaceQu'], axis ='columns')
test_df = test_df.drop(['Alley','PoolQC', 'Fence', 'MiscFeature','FireplaceQu'], axis ='columns')

Pour les données manquantes, nous avons fixé un seuil égal à 2% Nous supprimons donc toutes les lignes qui ont moins de 2% de données manquantes. 
Nous ferons donc ces transformations sur les variables : 

GarageArea	GarageArea	0.068540
KitchenQual	KitchenQual	0.068540
GarageCars	GarageCars	0.068540
TotalBsmtSF	TotalBsmtSF	0.068540
Exterior1st	Exterior1st	0.068540
Exterior2nd	Exterior2nd	0.068540
BsmtFinSF1	BsmtFinSF1	0.068540
BsmtFinSF2	BsmtFinSF2	0.068540
SaleType	SaleType	0.068540
BsmtUnfSF	BsmtUnfSF	0.068540
BsmtFullBath	BsmtFullBath	0.137080
Functional	Functional	0.137080
Utilities	Utilities	0.137080
BsmtHalfBath	BsmtHalfBath	0.137080
MSZoning	MSZoning	0.274160
MasVnrArea	MasVnrArea	1.028101
MasVnrType	MasVnrType	1.096642
Electrical	Electrical	0.068493
MasVnrArea	MasVnrArea	0.547945
MasVnrType	MasVnrType	0.547945

In [10]:
test_df.shape
train_df.shape

(1460, 76)